src/sdam/server.ts

```diff
@@ -297,24 +298,25 @@ export class Server extends TypedEventEmitter<ServerEvents> {
     }
 
     // Clone the options
-    const finalOptions = Object.assign({}, options, {
+    const options = {
+      ...paramOpts,
       wireProtocolCommand: false,
       directConnection: this.topology.s.options.directConnection
```

Comment:
why rename finalOptions, causing the parameter to need to be renamed?

Reply:
`conn = await this.pool.checkOut(options);` 

checkout is given the parameter options and command was given `finalOptions` which doesn't seem right, and the signal wasn't originally being passed to both correctly in an earlier iteration

Reply:
Why can't we just pass finalOptions in everywhere?

`paramopts` is strange - I've never seen this in JS before.  Alternatively, if you don't want to use finalOptions everywhere, could we just re-assign options?

```
options = { 
   ...options, 
  ...
}
```

Reply:
Yea happy to change it, just picked one way to solve it

Reply:
I think we actually don't need this code at all 🤦🏻 `wireProtocolCommand` doesn't seeem to be used anywhere. and `directConnection` is already passed in.

Reply:
directConnection is still needed here but wireProtocolCommand can easily be removed. much simpler :)